openpgp/x25519/x25519.go

```diff
@@ -0,0 +1,217 @@
+package x25519
+
+import (
+	"crypto/sha256"
+	"crypto/subtle"
+	"io"
+
+	"github.com/ProtonMail/go-crypto/openpgp/aes/keywrap"
+	"github.com/ProtonMail/go-crypto/openpgp/errors"
+	x25519lib "github.com/cloudflare/circl/dh/x25519"
+	"golang.org/x/crypto/hkdf"
+)
+
+const hkdfInfo = "OpenPGP X25519"
+const aes128KeySize = 16
+const PointSize = 32
+
+type PublicKey struct {
+	Point []byte
+}
+
+type PrivateKey struct {
+	PublicKey
+	Secret []byte
+}
+
+func NewPrivateKey(key PublicKey) *PrivateKey {
+	return &PrivateKey{
+		PublicKey: key,
+	}
+}
+
+// Validate validates that the provided public key matches
+// the private key.
+func Validate(pk *PrivateKey) (err error) {
+	var expectedPublicKey, privateKey x25519lib.Key
+	subtle.ConstantTimeCopy(1, privateKey[:], pk.Secret)
+	x25519lib.KeyGen(&expectedPublicKey, &privateKey)
+	if subtle.ConstantTimeCompare(expectedPublicKey[:], pk.PublicKey.Point) == 0 {
+		return errors.KeyInvalidError("x25519: invalid key")
+	}
+	return nil
+}
+
+// GenerateKey generates a new x25519 key pair
```

Comment:
```suggestion
// GenerateKey generates a new x25519 key pair.
```

Top-level comments must end with a dot

Reply:
3198ed3fada927f01bd289c9e3339891f395df17